src/acceleration/Acceleration.cpp

```diff
@@ -22,7 +22,8 @@ void Acceleration::applyRelaxation(double omega, DataMap &cplData)
     if (couplingData.timeStepsStorage().empty()) {
       continue;
     }
-
+    // @todo: This will apply the scaling to the sample at t=0 and t=1 when
+    // calling performAcceleration the first time. Is that intended?
```

Comment:
@BenjaminRodenberg is this correct?

Reply:
At `t=0` the value of the sample should be stationary from one iteration to the next, so the acceleration should not have a visible effect. But applying the acceleration in the first place also looks unnecessary to me.

Reply:
> With extraplation, these configuration files should also trigger some error. If not, then we have a bug in preCICE. Is this the case?

No, it has no effect

> But applying the acceleration in the first place also looks unnecessary to me.

Yes, how do you handle this in other places?

Reply:
Other accelerations do not yet support waveforms. I would actually try to avoid including the window start. Could lead to singular systems.

Reply:
I think you can copy-paste the "Don't use samples at the window start"-strategy pretty much from https://github.com/precice/precice/pull/1899. If this introduces too many changes here, let's try to merge https://github.com/precice/precice/pull/1899 first.

Reply:
I just took a closer look.

> Other accelerations do not yet support waveforms

That's only kind of true, as constant acceleration is supported and it also uses exactly the same functionality. So, this code section is actually part of https://github.com/precice/precice/pull/1794

> Could lead to singular systems.

Could you give an example of where this could happen? A cannot think of such an example as the `oldValue` will just be the same as before 

https://github.com/precice/precice/blob/586ac794e5e23bdc35aae623b919d9f260e5e474/src/acceleration/Acceleration.cpp#L29

> I think you can copy-paste the "Don't use samples at the window start"-strategy pretty much from https://github.com/precice/precice/pull/1899. If this introduces too many changes here, let's try to merge https://github.com/precice/precice/pull/1899 first.

The change here are not related to #1899 and I don't see a reason to connect these two things here.

Reply:
> Didn't you ask for a way how to not apply the acceleration to the sample at t=0

I was mainly wondering how you handle this elsewhere and the answer is 'we don't handle this until now'. I'm not familiar with the whole storage class, but I don't really understand why introducing a dedicated iterator range for all, but the first storage sample is not possible. We actually do this for the communication https://github.com/precice/precice/blob/586ac794e5e23bdc35aae623b919d9f260e5e474/src/utils/IntraComm.hpp#L37

Still, the PR here adds only test for a fix about the Aitken acceleration.

Reply:
> Could you give an example of where this could happen? A cannot think of such an example as the oldValue will just be the same as before

This can give problems for quasi-Newton, but not Aitken. Was only mention this as an answer to

> Yes, how do you handle this in other places?